2021/CifarelliDiFranco/README.md

```diff
@@ -0,0 +1,57 @@
+# DNSLA
+A short and simple software to analyze incorrect DNS responses developed by Marco Cifarelli and Leonardo Di Franco.
+
+## Preview
+This software mainly makes use of Tshark's ring buffer capture, pyshark's (which also is Tshark but is its python wrapper) file capture and TICK stack (Telegraf, Influx, Chronograf and Kapacitor) without Telegraf. Chronograf is the general manager of the stack and can be used after its activaton at localhost on port 8888: from now on connecting Kapacitor and Influx to Chronograf will be guided by the web page. Every part of the stack can be found easily at https://docs.influxdata.com/platform/install-and-deploy/install/oss-install/.
+
+## Functioning
+As soon as the software starts, Tshark is run as a subprocess (making use of python.subprocess.Popen so that the main process doesn't have to wait its termination), with a ring buffer (Tshark's ring buffer functioning can be found at https://www.wireshark.org/docs/man-pages/tshark.html) and its pid is taken and saved for later purposes. It will continuously capture incoming traffic related to the subnet identified by the IP that's over the specified network interface and will put a .pcap file into the _livecap_ directory every 10 seconds.
```

Comment:
La cattura live e' stata scartata per quale motivo?

Reply:
è stato scartato l'utilizzo di LiveCapture perchè con essa era necessario sincronizzare la struttura dati IP_resps e per farlo era necessario effettuare una lock per ogni pacchetto (inefficiente) oppure catturare ad intervalli di tempo costanti (col rischio di perdere dei pacchetti).
Comunque, la cattura con ring buffer è una cattura live che viene divisa in più pcap, quindi, non è stata scartata la cattura live in sè.

Reply:
il problema di questo metodo e' la sicurezza e ad ogni modo come vi avevo spiegato NON e' necessario sincronizzare i thread. Ad ogni modo va bene

Reply:
Su questo ha ragione prof. ma noi non abbiamo trovato un modo (per quanto potesse esser banale) di farlo senza usare sincronizzazioni e abbiamo ricorso a questo metodo che sembrava buono per noi.

Reply:
Ok, volta scelta progettuale, era solo per dirvi che la sincronizzazione non era necessaria e quindi il vostro era un non problema.